src/utils/DependencyUtils.ts

```diff
@@ -11,3 +29,65 @@ export const splitToNameAndVersion = (dependencyString: string): [string, string
     const fullName = parts.join('-');
     return [fullName, version];
 };
+
+export async function getFullDependencyList(
+    combos: ThunderstoreCombo[],
+    game: Game,
+    installedMods: ManifestV2[],
+    modpackDependencyStrategy: ModpackDependencyStrategy
+) {
+    const originalCombos = [...combos];
+    for (const combo of originalCombos) {
+        const isModpack = combo.getMod().getCategories().some(value => value === "Modpacks");
+        const dependencyVersionStrategy = isModpack ? modpackDependencyStrategy : ModpackDependencyStrategy.USE_LATEST_VERSION;
+
+        await buildDependencySet(combo.getVersion(), game, combos, dependencyVersionStrategy);
+
+        // By default this filters out mods that are already installed.
+        // However, in the case of modpacks we don't want to filter them out,
+        // because modpacks (by default*) want the dependencies at specific versions.
+        // *(unless the modpackDependencyStrategy is set to USE_LATEST_VERSION)
```

Comment:
Uh, this last line confuses me. If things are as the comment suggests, should the if-block then check `dependencyVersionStrategy` rather than `!isModPack`?

Reply:
The comment should now be very descriptive